src/dune/scheduler.ml

```diff
@@ -214,10 +204,13 @@ end = struct
     if not avail then Condition.signal cond;
     Mutex.unlock mutex
 
-  let send_dedup file =
+  (* FIXME: this is really not ideal, but we pack the Caching in the event all
+     the way through since Scheduler cannot read it directly from the file
+     system because of circular dependencies. *)
```

Comment:
What about dropping these events altogether right from the source when the mode is `copy`? Ideally, the daemon wouldn't even send them if it can figure out that the client is running in copy mode

Reply:
I can skip sending these in copy mode, but I'd still like to delegate deduplication details to the cache so it can do whatever it wants.

Reply:
Ok. FTR, passing along the caching seems fine to me